frontend/src/__tests__/integration/pages/pipelines/PipelineDetails.spec.ts

```diff
@@ -0,0 +1,22 @@
+import { test, expect } from '@playwright/test';
+import { navigateToStory } from '~/__tests__/integration/utils';
+
+test('Test topology renders', async ({ page }) => {
+  await page.goto(navigateToStory('pages-pipelines-pipelinedetails', 'default'));
+
+  // wait for page to load
+  await page.waitForSelector('text=test-pipeline');
+
+  // check the topology renders
+  expect(page.getByText('flip-coin')).toBeTruthy();
+  expect(page.getByText('random-num')).toBeTruthy();
+  expect(page.getByText('print-message')).toBeTruthy();
+
+  // check topology has tasks
+  await page.getByText('print-msg', { exact: true }).click();
+  await page.getByText('$(tasks.random-num.results.Output)').click();
+  await page.getByLabel('Close drawer panel').click();
+  await page.getByText('flip-coin').click();
+  await page.getByText('/tmp/outputs/Output/data').click();
+  await page.getByLabel('Close drawer panel').click();
```

Comment:
Are these all action items? They seem like text... do you need to click all of them?

Reply:
it is pressing on nodes then checking that the drawer opened with the correct text

Reply:
If I am understanding this correctly -- This is not ideal... you're opening up a node (`print-msg`), then making use of a click action to verify the **text** (eg. `'$(tasks.random-num.results.Output)'`) is on the screen.  That is very misleading. @christianvogt FYI.

We can go ahead with this if we want to merge this and use it in your work. I'm definitely going to need this to be improved as we move to Cypress.

Reply:
At the very least we should be using role based selectors.

Reply:
there was not a role i could use there as far as i know. I did remove the clicking of the text tho